chain/network/src/peer/peer_actor.rs

```diff
@@ -545,6 +546,82 @@ impl PeerActor {
             }
         }
     }
+
+    /// Check whenever `Codec::decode` method banned peer for message being too long.
+    /// If so bans peers
+    /// or returns the decoded message.
+    fn check_if_decoder_banned_peer(
+        &mut self,
+        msg: Result<Vec<u8>, ReasonForBan>,
+        ctx: &mut Context<PeerActor>,
+    ) -> Option<Vec<u8>> {
+        let msg = match msg {
+            Ok(msg) => msg,
+            Err(ban_reason) => {
+                self.ban_peer(ctx, ban_reason);
+                return None;
+            }
+        };
+        Some(msg)
+    }
+
+    /// Update stats when receiving msg
+    fn update_stats_on_receiving_message(&mut self, msg_len: usize) {
+        metrics::PEER_DATA_RECEIVED_BYTES.inc_by(msg_len as u64);
+        metrics::PEER_MESSAGE_RECEIVED_TOTAL.inc();
+        self.tracker.increment_received(msg_len as u64);
+    }
+
+    /// Check whenever we exceeded number of transactions we got since last block.
+    /// If so, drop the transaction.
+    fn should_we_drop_msg_without_decoding(&mut self, msg: &Vec<u8>) -> bool {
+        if codec::is_forward_transaction(&msg).unwrap_or(false) {
+            let r = self.txns_since_last_block.load(Ordering::Acquire);
+            if r > MAX_TRANSACTIONS_PER_BLOCK_MESSAGE {
+                return true;
+            }
+        }
+        false
+    }
+
+    /// Decodes message using `BorshDeserialize
+    /// Somehow checks if there is a version mismatch. How exactly you ask, that's a great question?
+    /// TODO: Figure this out.
+    fn try_to_decode_to_peer_message(&mut self, msg: &Vec<u8>) -> Option<PeerMessage> {
+        let peer_msg = match bytes_to_peer_message(&msg) {
```

Comment:
Similarly, I'd probably keep the `match` at the cal-site, and extracted only the `Err(err) =>` branch into a function.

Reply:
@matklad Thanks, that's even better.